### PR TITLE
[PATCH API-NEXT v3] Implement checksum insertion override for dpdk pktio

### PR DESCRIPTION
**Pull request by bogdanPricope**
Implement checksum insertion override generic functions and processing for dpdk pktio.

### CODE REVIEW

File: platform/linux-generic/odp_packet.c [hunk 1]
```diff
@@ -1259,6 +1259,22 @@ int odp_packet_l4_offset_set(odp_packet_t pkt, uint32_t offset)
...
 
+void odp_packet_l3_chksum_insert(odp_packet_t pkt, int l3)
+{
+	odp_packet_hdr_t *pkt_hdr = packet_hdr(pkt);
+
+	pkt_hdr->p.output_flags.l3_chksum_set = 1;
+	pkt_hdr->p.output_flags.l3_chksum = l3;
+}
+
+void odp_packet_l4_chksum_insert(odp_packet_t pkt, int l4)
```
Comment (muvarov) [inline platform/linux-generic/odp_packet.c]: maybe it's reasonable to change api at the same time to:
void odp_packet_l3_chksum_insert(odp_packet_t pkt,  odp_bool_t enable)
?
this int l3, int l4 is confusion.
@psavol ?

**Comment by lumag** [comment]
This should be reworked so that linux-generic provides default support for checksumming, if pktio doesn't have optimized implementation.

**Comment by bogdanPricope** [comment]
@lumag  "Calling this function is always allowed but the checksum will not be
inserted if the packet is output through a pktio that does not have
the relevant pktout chksum bit set in the pktio capability."

Let’s start from the assumption that API was discussed and agreed on at the time was proposed and merged.

**Comment by lumag** [comment]
@bogdanPricope sorry, I probably was confused between different API calls.
Yes, this looks good then.

File: platform/linux-generic/odp_packet.c (hunk 1 above)
Comment (Bill-Fischofer-Linaro) [reply platform/linux-generic/odp_packet.c]: Agree with @muvarov 

Perhaps rework this in light of PR #264?

**Comment by muvarov** [comment]
@lumag do you want to add your Review-by ?

**Comment by lumag** [comment]
Reviewed-by: Dmitry Eremin-Solenikov

**Comment by Bill-Fischofer-Linaro** [comment]
@bogdanPricope Please rebase to resolve conflicts in `odp_packet.c`

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**